File: backend/src/routes/cidades.js
const router = require('express').Router();
const auth = require('../middleware/auth');
const supabase = require('../db');

router.get('/', auth, async (req, res) => {
  try {
    const { data, error } = await supabase.from('cidades').select('*').order('nome');
    if (error) return res.status(400).json({ error: error.message });
    res.json(data);
  } catch (e) { res.status(500).json({ error: e.message }); }
});

router.post('/', auth, async (req, res) => {
  try {
    const { data, error } = await supabase.from('cidades').insert(req.body).select().single();
    if (error) return res.status(400).json({ error: error.message });
    res.status(201).json(data);
  } catch (e) { res.status(500).json({ error: e.message }); }
});

module.exports = router;

File: backend/app/routes/cidades.py
from fastapi import APIRouter, Depends
from app.db import get_supabase
from app.auth import get_current_user

router = APIRouter()

@router.get("/")
async def list_cidades(user=Depends(get_current_user)):
    return get_supabase().table("cidades").select("*").order("nome").execute().data

@router.post("/")
async def create_cidade(body: dict, user=Depends(get_current_user)):
    res = get_supabase().table("cidades").insert(body).execute()
    return res.data[0] if res.data else {}
